extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,212 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-story-quiz-1.0.css';
+import {createShadowRootWithStyle} from './utils';
+import {dev} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {toArray} from '../../../src/types';
+
+/** @const {!Array<string>} */
+const answerChoiceOptions = ['A', 'B', 'C', 'D'];
+
+/** @const {string} */
+const TAG = 'amp-story-quiz';
+
+/**
+ * Generates the template for the quiz
+ * @param {HTMLElement} element
+ * @return {HTMLElement}
+ */
+const buildQuizTemplate = element => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amp-story-quiz-container">
+      <div class="i-amp-story-quiz-prompt-container"></div>
+      <div class="i-amp-story-quiz-option-container"></div>
+    </div>
+  `;
+};
+
+/**
+ * Generates the template for each option
+ *
+ * @param {HTMLElement} option
+ * @return {HTMLElement}
+ */
+const buildOptionTemplate = option => {
+  const html = htmlFor(option);
+  return html`
+    <span class="i-amp-story-quiz-option">
+      <span class="i-amp-story-quiz-answer-choice"></span>
+    </span>
+  `;
+};
+
+export class AmpStoryQuiz extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {boolean} */
+    this.hasReceivedResponse_ = false;
+
+    /** @private {?ShadowRoot|HTMLDivElement} */
+    this.quizElt_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.quizElt_ = buildQuizTemplate(this.element);
+    this.attachContent_();
+    this.attachOptionActionHandlers_();
+    createShadowRootWithStyle(this.element, this.quizElt_, CSS);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @return {?ShadowRoot|HTMLDivElement}
+   */
+  getQuizElement() {
+    return this.quizElt_;
+  }
+
+  /**
+   * Finds the prompt and options content
+   * and adds it to the quiz element.
+   * @private
+   */
+  attachContent_() {
+    // TODO(jackbsteinberg): Optional prompt behavior must be implemented here
+    const prompt = this.element.children[0];
+    // First child must be heading h1-h3
+    if (!['h1', 'h2', 'h3'].includes(prompt.tagName.toLowerCase())) {
+      dev().error(
+        TAG,
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    prompt.classList.add('i-amp-story-quiz-prompt');
```

Comment:
How do you ensure there's nothing harmful within that `<hn>` tag?
Why do we support `<h3 style="foo: bar;">...</h3>` for the prompt and not the options?

Reply:
Changed this to stripping the textContent and adding to a new element with the same tagName